src/System.IO.Abstractions.TestingHelpers/MockDirectoryData.cs

```diff
@@ -25,9 +21,16 @@ public new DirectorySecurity AccessControl
             {
                 // DirectorySecurity's constructor will throw PlatformNotSupportedException on non-Windows platform, so we initialize it in lazy way.
                 // This let's us use this class as long as we don't use AccessControl property.
-                return accessControl ?? (accessControl = new DirectorySecurity());
+                var value = Extensibility.RetrieveMetadata("AccessControl:DirectorySecurity");
+                if (value is not DirectorySecurity directorySecurity)
+                {
+                    directorySecurity = new DirectorySecurity();
+                    Extensibility.StoreMetadata("AccessControl:DirectorySecurity", directorySecurity);
+                }
+
+                return directorySecurity;
```

Comment:
May it is just me not being used to the ` value is not DirectorySecurity directorySecurity`, but I find this much more readable. I was staring for over a minute on the code to see were `directorySecurity` is defined or if it is a global variable. [ Feel free to ignore this style, very likely just personal taste, but I want to toss it out here ...]
```suggestion
                var value = Extensibility.RetrieveMetadata("AccessControl:DirectorySecurity") as DirectorySecurity;
                if (value == null)
                {
                    directorySecurity = new DirectorySecurity();
                    Extensibility.StoreMetadata("AccessControl:DirectorySecurity", directorySecurity);
                }

                return directorySecurity;
```

Reply:
Will change it as you suggest. I am just used to this new style, but can understand keeping it simple :-)